### DISCUSSION TITLE
Snapshot interpolation only works when I switch tabs for 10 seconds

### DISCUSSION THREAD

**Discussion by Seclusionist** [geckosio/geckos.io#244] [Help / Questions] [answered]
Hi there, I'm running into strange behavior: The snapshot interpolation code only works correctly when I switch tabs for 10 seconds and then switch back to the game. So I will move and then switch tabs for 10 seconds and come back and the code will be working correctly. What exactly could cause this to be the case?

**Comment by Quitalizner** [comment, marked as answer]
When you switch tabs, the browser stops sending inputs to the server. All your inputs will get bunched up. I believe the same happens for the data sent from the server to the browser

**Reply by Seclusionist** [reply]
I appreciate your thoughts on this, I've read over it a few days now trying to tinker with it more and am still unable to resolve this. Do you have any idea how I can use your information to fix this?

**Reply by yandeu** [reply]
Maybe this helps:
https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState

**Reply by Seclusionist** [reply]
I'm going to take a look and play with that for a while and see what happens, thank you yandeu very much

**Reply by Seclusionist** [reply]
@yandeu @Quitalizner I found out that it was an issue on my own end on my server side :smile: It is working perfect now without having to switch tabs. Thank you for both of your help :relaxed: What a major relief
